util/http/client/client_test.go

```diff
@@ -25,7 +27,40 @@ import (
 	. "github.com/smartystreets/goconvey/convey"
 )
 
-func TestHTTPClient_Fetch(t *testing.T) {
+func TestClient(t *testing.T) {
+	Convey("User-Agent should match Git version", t, func() {
+		var gotUserAgent string
+
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			td, err := json.Marshal(map[string]string{
+				"foo": "bar",
+			})
+			if err != nil {
+				panic(err)
+			}
+
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			w.Write(td)
+
+			gotUserAgent = r.UserAgent()
+		}))
+		defer ts.Close()
+
+		u, err := url.Parse(ts.URL)
+		if err != nil {
+			panic(err)
+		}
+
+		if err := Fetch(http.DefaultClient, *u, nil); err != nil {
+			panic(err)
+		}
+
+		So(gotUserAgent, ShouldContainSubstring, "com.mesosphere.dcos-metrics/")
```

Comment:
I suppose a hacky way of testing that the User-Agent was modified successfully during the build process would be to expose the user agent when calling `-version` on the command line, but I'd like to avoid that if I can.